### ISSUE TITLE
Pipelines: End-to-end Tutorial on Agile Stacks

### ISSUE THREAD

**Issue by mameshini** [kubeflow/website#799]
Provide an end-to-end tutorial for Kubeflow Pipelines available as a pipeline template via Agile Stacks.  A well-organized machine learning project template should modularize data processing, model definition, model training, validation, and inference tasks.  We are using this tutorial to demonstrate how to share code, data sets, and other artifacts between notebooks and pipeline steps. It can be deployed on GCP, AWS, and Azure.  

In the future, add more end-to-end tutorials, as suggested in #737. 

**Overview**
This tutorial will introduce you to the basic workings of Kubeflow and machine learning pipelines, and provide you with a Kubeflow pipeline template that you can use to jump start your own machine learning projects.  Based on the pipeline templates, we will implement several Kubeflow pipelines for training, monitoring, and deployment of deep learning models. We will use Kubeflow, Keras, and Seldon to implement end-to-end ML pipelines that can run on AWS, GCP, Azure, or on-premise hardware. We will also cover how you can use Kubeflow pipelines to continuously deploy models into production.  

We will be following these steps during the tutorial:

- Download and store training data in Minio buckets or shared storage.
- Implement steps for cleaning, labeling, and pre-processing training data sets. 
- Build, train, and optimize models directly from Jupyter notebooks.
- Train Seq2Seq NLP model in containers using multiple CPUs or GPUs.
- Deploy your machine learning models as REST services using Seldon.
- Reduce cloud cost with automation, autoscaling, spot instances, preemptible instances.
- Compare results of experiments using model monitoring and experiment tracking tools.
- Automate hyperparameter optimization, model versioning, and deployment.
- Create a web application for sending prediction requests to the deployed models using Seldon.
- Continuously deploy models and pipelines into production.
- Monitor how the model performs on real life data.

**Comment by sarahmaddox** [comment]
This looks excellent @mameshini. Will you tackle this during the upcoming doc sprint? If so, I'll add it to the doc-sprint label and Kanban board.

**Comment by mameshini** [comment]
@sarahmaddox Yes please add doc-sprint label to this issue.  The goal is to have it completed prior to the upcoming doc sprint so we can just review, fine tune, and publish it.

**Comment by sarahmaddox** [comment]
Great, thanks @mameshini
